docs/scaling/SYSTEM.md

```diff
@@ -0,0 +1,190 @@
+# Knative Serving Autoscaling System
+
+This document describes the Knative Serving Autoscaling system and goes into the
+details of how autoscaling is implemented in Knative Serving. It describes:
+
+- The components that form the Knative Serving Autoscaling system
+- The APIs at play that control the behavior of said components
+- The in-depth control- and data-flows happening in various scenarios like
+  scaling up/down and to/from zero
+
+## Components
+
+The autoscaling system consists of a few “physical” and logical components that
+are briefly defined here. Understanding what they are, where they are deployed
+and what they do will greatly help to understand the control- and data-flows.
+The components mentioned might do more things than outlined here. This document
+is going to stick only to the details that affect the autoscaling system.
+
+### Queue Proxy
+
+The **queue-proxy** is a sidecar container that is deployed alongside the
+user-container in **each user pod**. Each request sent to an instance of the
+application is going via the queue-proxy first, hence the “proxy” in its name.
+
+The queue-proxy’s main purpose is to **measure** and **limit concurrency** to
+the user’s application. If a revision defines a concurrency limit of 5, the
+queue-proxy makes sure that no more than 5 requests reach the application’s
+instance at once. If there are more requests being sent to it than that, it will
+queue them locally, hence the “queue” in its name. The queue-proxy also measures
+the incoming request load and reports the **average concurrency** and **requests
+per second** on a separate port.
+
+### Autoscaler
+
+The **autoscaler** is a stand-alone pod consisting of three main components:
+
+1. The PodAutoscaler reconciler
+2. The **Collector**
+3. The **Decider**
+
+The PodAutoscaler reconciler makes sure that any changes to PodAutoscalers (see
+API section) are correctly picked up and reflected in either the Decider, the
+Collector or both.
+
+The collector is responsible for collecting metrics from the queue-proxies on
+the application instances. To do so, it scrapes their internal metrics endpoint
+and sums them up to get a metric representative of the entire system. For
+scalability, only a sample of all application instances is scraped and the
+received metrics are extrapolated to the entire cluster.
+
+The decider gets all metrics available and decides how many pods the
+application’s deployment should be scaled to. Basically, the equation to do that
+is `want = concurrencyInSystem/targetConcurrencyPerInstance`.
+
+In addition to that, it adjusts the value for a maximum scale up/down rate and
+the min- and max-instances settings on the revision.
```

Comment:
I think here's the place to inject "proxy mode for scaled up resource" with some reference to tbc.